daemon/k8s/pods.go

```diff
@@ -0,0 +1,135 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package k8s
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/cilium/hive/cell"
+	"github.com/cilium/hive/job"
+	"github.com/cilium/statedb"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/util/duration"
+
+	"github.com/cilium/cilium/pkg/k8s"
+	"github.com/cilium/cilium/pkg/k8s/client"
+	slim_corev1 "github.com/cilium/cilium/pkg/k8s/slim/k8s/api/core/v1"
+	"github.com/cilium/cilium/pkg/k8s/utils"
+	nodeTypes "github.com/cilium/cilium/pkg/node/types"
+	"github.com/cilium/cilium/pkg/time"
+)
+
+// LocalPod is Cilium's internal model of the pods running on this node.
+type LocalPod struct {
+	*slim_corev1.Pod
+
+	// UpdatedAt is the time when [LocalPod] was last updated, e.g. it
+	// shows when the pod change was received from the api-server.
+	UpdatedAt time.Time `json:"updatedAt" yaml:"updatedAt"`
+}
+
+func (p LocalPod) TableHeader() []string {
+	return []string{
+		"Name",
+		"HostNetwork",
+		"PodIPs",
+		"Containers",
+		"Phase",
+		"Age",
+	}
+}
+
+func (p LocalPod) TableRow() []string {
+	var podIPs []string
```

Comment:
Nit: `podIPs := make([]string, 0, len(p.Status.PodIPs)`
Same for `containers` and `ports`